### PR TITLE
MULE-19402: FeatureFlaggingService improvement

### PR DESCRIPTION
**Pull request by IvanAndresFritzler**

### CODE REVIEW

**Comment by IvanAndresFritzler** [comment]
--validate

**Comment by IvanAndresFritzler** [comment]
--validate

File: core/src/main/java/org/mule/runtime/core/api/config/FeatureFlaggingRegistry.java [hunk 1]
```diff
@@ -70,35 +71,69 @@ private FeatureFlaggingRegistry() {}
...
   public void registerFeature(Feature feature, Predicate<MuleContext> condition) {
+    validate(feature, condition);
+    Predicate<MuleContext> added = configurations.putIfAbsent(feature, condition);
+    if (added != null) {
+      throw new IllegalArgumentException(format(FEATURE_ALREADY_REGISTERED, feature));
+    }
+  }
+
+  /**
+   * Registers a {@link Predicate} associated with a String which represents a given feature. The {@link Predicate} will be
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/api/config/FeatureFlaggingRegistry.java:86]: what do you mean by `associated with a String`? where is a String here?

File: core/src/main/java/org/mule/runtime/core/api/config/FeatureFlaggingRegistry.java [hunk 2]
```diff
@@ -70,35 +71,69 @@ private FeatureFlaggingRegistry() {}
...
+   * evaluated at deployment time, exposing all the features through a per-application overridden {@link FeatureFlaggingService}
+   *
+   * @see FeatureFlaggingService
+   *
+   * @param feature   Name representing the registered feature
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/api/config/FeatureFlaggingRegistry.java:91]: `name` is misleading

File: core/src/main/java/org/mule/runtime/core/internal/config/FeatureFlaggingServiceBuilder.java [hunk 3]
```diff
@@ -19,56 +19,123 @@
 import org.mule.runtime.api.config.Feature;
 import org.mule.runtime.core.api.MuleContext;
 import org.mule.runtime.api.config.FeatureFlaggingService;
+import org.mule.runtime.core.api.config.FeatureContext;
 import org.slf4j.Logger;
 
 /**
- * This builder creates a {@link FeatureFlaggingService}.
+ * <p>
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/config/FeatureFlaggingServiceBuilder.java:26]: remove this separator at the beginning

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/api/config/FeatureFlaggingRegistry.java (hunk 1 above)
Comment (IvanAndresFritzler) [reply core/src/main/java/org/mule/runtime/core/api/config/FeatureFlaggingRegistry.java:86]: I just copied the deprecated method documentation, but this is a confusing text indeed... will update both.

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**